### PR TITLE
translations: adds editor translations support

### PR DESCRIPTION
**Pull request by jma**
* Removes JSONSchema backend on the fly translations.
* Removes legacy `validationMessage` JSONSchema` property.
* Adds suffix to the validation messages keys.
* Fixes missing punctuation in the validation messages.
* Cleans babel configuration.
* Removes legacy schema form code.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- FixesL https://tree.taiga.io/project/rero21-reroils/task/1456?kanban-status=1224894

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* https://github.com/rero/ng-core/pull/187
* https://github.com/rero/rero-ils-ui/pull/277

## How to test?

- Go to the professionnal interface
  - the language menu should have the same behaviour than the public interface
  - test the editor with several resources, switch the language, the label, the description the placeholder and the validation should be translated.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 1]
```diff
@@ -1152,14 +1152,18 @@
...
               "invalid",
               "cancelled",
               "invalid or cancelled"
             ],
             "form": {
-              "hide": true
+              "hide": true,
+              "validation": {
+                "messages": {
+                  "patternMessage": "ISBN/ISSN status should be selected in the list below."
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json:1164]: Is it a closed list? If yes, maybe we don't even need a `patternMessage`.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 2]
```diff
@@ -1168,14 +1168,18 @@
...
               "invalid",
               "cancelled",
               "invalid or cancelled"
             ],
             "form": {
-              "hide": true
+              "hide": true,
+              "validation": {
+                "messages": {
+                  "patternMessage": "ISBN/ISSN status should be selected in the list below."
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json:1180]: Same as precedent question.

File: rero_ils/modules/patrons/jsonschemas/patrons/patron-v0.0.1.json [hunk 3]
```diff
@@ -64,7 +64,7 @@
       "form": {
         "validation": {
           "messages": {
-            "pattern": "Should be in the ISO 8601, YYYY-MM-DD."
+            "patternMessage": "Should be in the ISO 8601, YYYY-MM-DD."
```
Comment (iGormilhit) [inline rero_ils/modules/patrons/jsonschemas/patrons/patron-v0.0.1.json:67]: Maybe the same message as in https://github.com/rero/rero-ils/pull/1021/files#diff-012a8082b0a21ed9db84b8553aac2c6eR80 ?

File: rero_ils/views.py [hunk 4]
```diff
@@ -283,14 +282,7 @@ def prepare_jsonschema(schema):
         del schema['$schema']
     schema['required'].remove('pid')
     keys = current_app.config['RERO_ILS_BABEL_TRANSLATE_JSON_KEYS']
```
Comment (lauren-d) [inline rero_ils/views.py:284]: It seems that `keys` is not used anymore. It can be removed.

Comment (lauren-d) [inline rero_ils/views.py:284]: don't forget to remove that 😉

**Pull request merged**